narwhals/_dask/namespace.py

```diff
@@ -135,6 +137,41 @@ def any_horizontal(self, *exprs: IntoDaskExpr) -> DaskExpr:
     def sum_horizontal(self, *exprs: IntoDaskExpr) -> DaskExpr:
         return reduce(lambda x, y: x + y, parse_into_exprs(*exprs, namespace=self))
 
+    def concat(
+        self,
+        items: Iterable[DaskLazyFrame],
+        *,
+        how: str = "vertical",
+    ) -> DaskLazyFrame:
+        import dask.dataframe as dd  # ignore-banned-import
+
+        if len(list(items)) == 0:
+            msg = "No items to concatenate"
+            raise ValueError(msg)
+        native_frames = [i._native_frame for i in items]
+        axis: int
+        if how == "vertical":
+            all_columns = set(chain(*[i.columns for i in native_frames]))
+            if not all(set(i.columns) == all_columns for i in native_frames):
+                msg = "unable to vstack with non-matching columns"
+                raise TypeError(msg)
+            axis = 0
+        elif how == "horizontal":
+            if len({len(i) for i in native_frames}) != 1:
```

Comment:
`len` will trigger computations and we would rather avoid that.
I am not sure if `dask_expr._expr.are_co_aligned` applied to dataframes, but if it does, I think that the following check could be all we need:

```suggestion
            if not get_dask_expr()._expr.are_co_aligned(*native_frames):
```

Reply:
Looks like `are_co_aligned` doesn't work (throws an error when it shouldn't in testing). I think its because `are_co_aligned` is checking a bit more than things just having the same length, which actually makes me wonder what horizontal concatenation would do if dataframes had the same length but different partitioning (I think it would probably error but not get caught with my check).

I added in the check as a convenience though, it's probably fine to take out (dask will just raise an error for invalid vertical concatenation later on) - what do you think?

Reply:
I've taken *out* the check for now, which I've convinced myself it a good move because:
- I don't think my `len` comparison is really covering all non-valid cases
- It avoids the additional compute